src/main/java/org/openstreetmap/atlas/checks/validation/intersections/LineCrossingWaterBodyCheck.java

```diff
@@ -450,8 +466,13 @@ private boolean canCrossWaterBody(final AtlasItem crossingItem,
                     final PolyLine waterbodyComponentGeometry = member.getEntity() instanceof Area
                             ? new Polygon((Area) member.getEntity())
                             : new PolyLine((LineItem) member.getEntity());
-                    return new Tuple<>(waterbodyComponentGeometry,
-                            intersectingFeature.intersections(waterbodyComponentGeometry));
+                    Set<Location> intersectionLocations = intersectingFeature
```

Comment:
It looks like you're adding to the CheckFlag early here before some filtering happens. We should save the Set<Location> defined by the return value starting [here](https://github.com/osmlab/atlas-checks/pull/515/files#diff-c3c493b5180ef0e0d356fbced682bc317cafd2f0cb8a7f2084cfbe2495d051e9R456) to a variable and just add it to the CheckFlag, then return that variable

Reply:
addressed!